File: src/compiler/typescript_to_rust/generators/ic_object/functions/arg_data_raw.ts
import { Rust } from '../../../../../types';

export function generateIcObjectFunctionArgDataRaw(): Rust {
    return /* rust */ `
        fn _azle_ic_arg_data_raw(
            _this: &boa_engine::JsValue,
            _aargs: &[boa_engine::JsValue],
            _context: &mut boa_engine::Context
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            Ok(ic_cdk::api::call::arg_data_raw().azle_into_js_value(_context))
        }
    `;
}

File: src/compiler/rust/canister/src/ic/arg_data_raw.rs
use wasmedge_quickjs::{Context, JsFn, JsValue};

pub struct NativeFunction;
impl JsFn for NativeFunction {
    fn call(context: &mut Context, _this_val: JsValue, _argv: &[JsValue]) -> JsValue {
        context
            .new_array_buffer(&ic_cdk::api::call::arg_data_raw())
            .into()
    }
}
